### ISSUE TITLE
How to delete unsed sst in Rockdb

### ISSUE THREAD

**Issue by stelios6** [arangodb/arangodb#7442]
## My Environment

* __ArangoDB Version__:        3.4 RC3
* __Storage Engine__:             RocksDB -
* __Deployment Mode__:       Single Server 
* __Deployment Strategy__:   
* __Configuration__:               
* __Infrastructure__:               
* __Operating System__:        | Windows server 2012
* __Used Package__:              

## Steps to reproduce

1. I have sst file in total of 40GB but when exporting the arango db (using arangodump ) the size is much less. How can I delete unused sst in order not to use disk space?
2. 
3.

**Comment by stelios6** [comment]
Any news on the above?

**Comment by graetzer** [comment]
do you have many indexes on your data ?

**Comment by stelios6** [comment]
Apart from the default key yes one field.

Stelios



Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by graetzer** [comment]
quick question, can your manually start the compaction your server ? You can do so using the `db.yourcollection.compact()` command in the emergency console (start arangod with --console)

**Comment by stelios6** [comment]
Can you provide more info pls. I have started the arangod using the --console command but the command you have provided does not work. I can not find any documentation on the Internet on how to execute this command.

**Comment by Simran-B** [comment]
Please try `require("internal").db.yourcollection.compact()`. This is only available in the arangod emergency console, not in arangosh,

**Comment by stelios6** [comment]
Still the same problem. I get the error Javascript exception :TypeError: Cannot read property 'compact' of undefined.

**Comment by graetzer** [comment]
You have to replace “yourcollection“ with the name your collections actually has

**Comment by stelios6** [comment]
Yes i did and I get the above error. Do I have to change to the specific db since the default is the root one? Is there another way to run the compact command?

**Comment by graetzer** [comment]
Oh yes, you need to run `db.useDatabase(‘name’)`

**Comment by graetzer** [comment]
Sorry I missed the underscore `db._useDatabase(‘name’)`

**Comment by stelios6** [comment]
Ok now the command useDatabase works
Once i execute the second command it does not seem to be working.

require("internal").db.mycollectionname.compact(). I have tried with underscore as well without any luck.





Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by Simran-B** [comment]
Do I understand correctly, that you don't see any error, but the sst files are still there after the execution of the compact() command?

**Comment by stelios6** [comment]
I do see an error. It seems the function compact() does not exist since it is not understood by the console. Can I execute the command without running arangod in console mode since I have to stop the service every time and run in console mode?

**Comment by graetzer** [comment]
```
const db = require('internal').db;
db._useDatabase('dbname')
db.yourcollection.compact()
```

**Comment by stelios6** [comment]
Ok now the commands are executed successfully. The command is still being executed of course since the collection has more than 100GB. Just a question. Instead the folder of the rockdb to get smaller in MBs it is getting bigger. Is that the expected behavour?



Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by Simran-B** [comment]
While compaction runs I would expect it to grow indeed, because it may need to write new files and can't delete the old ones yet (at least that's what I assume).

**Comment by Simran-B** [comment]
@stelios6 There is a pending change to expose .compact() in arangosh if you are interested:
https://github.com/arangodb/arangodb/pull/8468/files

**Comment by Simran-B** [comment]
@stelios6 Starting with v3.4.5, there will be a compact() method in arangosh too:
https://github.com/arangodb/arangodb/pull/8471
